sample-functions/ApiKeyProtected-Secrets/handler.go

```diff
@@ -8,10 +8,21 @@ import (
 	"strings"
 )
 
+func getAPISecret(secretName string) (secretBytes []byte, err error) {
```

Comment:
I would be inclined to swallow the error logging into the function rather than have the user deal with it when getting the secret:
```
// getSecret returns the secret found at the first valid secret mountpoint
func getSecret(name string) []byte {
	mounts := []string{"/var/openfaas/secrets/", "/run/secrets/"}
	var b []byte
	var err error
	for _, m := range mounts {
		if b, err = ioutil.ReadFile(m + name); err == nil {
			return b
		}
	}
	log.Fatal(err)
	return nil
}
```
```
secretBytes := getSecret("secret_api_key")
```